behave/features/steps/regex_steps.py

```diff
@@ -33,23 +27,69 @@ def step_impl_self_employed(context, optional):
 
 @step("I do (?P<optional>not )?have a partner")
 def step_impl_has_partner(context, optional):
-    state = "true"
-    if optional:
-        state = "false"
-    radio_input = context.helperfunc.find_by_css_selector(
-        f"input[name='your_details-has_partner'][value='{state}']"
-    )
-    radio_input.click()
-    assert radio_input.get_attribute("checked") == "true"
+    assert_select_radio_button(context, optional, "your_details-has_partner")
 
 
 @step("I am (?P<optional>not )?aged 60 or over")
 def step_impl_over_sixty(context, optional):
-    state = "true"
+    assert_select_radio_button(context, optional, "your_details-older_than_sixty")
+
+
+@step("I am (?P<optional>not )?aged 17 or under")
+def step_impl_under_eighteen(context, optional):
+    assert_select_radio_button(context, optional, "your_details-is_you_under_18")
+
+
+@step("I do (?P<optional>not )?receive money on a regular basis")
+def step_impl_receive_money_regularly(context, optional):
+    assert_select_radio_button(
+        context, optional, "your_details-under_18_receive_regular_payment"
+    )
+
+
+@step(
+    "I do (?P<optional>not )?have savings, items of value or investments totalling £2500 or more?"
+)
+def step_impl_have_savings_items_over_two_thousand_five_hundred(context, optional):
+    assert_select_radio_button(context, optional, "your_details-under_18_has_valuables")
+
+
+@step("I can not answer the following 17 or under questions")
+def step_impl_under_18_no_follow_up_questions_fail(context):
+    has_valuables_radio_buttons = context.helperfunc.driver().find_elements_by_xpath(
+        "//input[@name='your_details-under_18_has_valuables']"
+    )
+    regular_payment_radio_buttons = context.helperfunc.driver().find_elements_by_xpath(
+        "//input[@name='your_details-under_18_receive_regular_payment']"
+    )
+    assert (
+        len(has_valuables_radio_buttons) == 0
+    ), "Expected under 18 has valuables question to be hidden"
```

Comment:
this code could be reused for line 75 on

Reply:
I refactored this in an admittedly weird way. Let me know what you think.